src/platform/packages/shared/kbn-monaco/src/languages/esql/lib/converters/suggestions.ts

```diff
@@ -42,9 +53,26 @@ export function wrapAsMonacoSuggestions(
         hasAnIncompleteSuggestion = true;
       }
 
+      let insertText = text;
+
+      if (replaceParamsWithDefaults) {
+        // Replace ?_tstart and ?_tend parameters with default/dynamic values
+        // Note: dynamic end timestamp evaluated at suggestion generation time
+        // Precompute once per suggestion so multiple occurrences share identical timestamps
+        const dynamicStart = getDynamicTstartParam();
```

Comment:
This is mostly a nit but calling them differently means that there will be a small time gap as you have 2 Date.now()

I would consider merging them to one. Something like

```
const TIME_WINDOW = 15 * 60 * 1000;

function geTimeParamsValues(): { start: string; end: string } {
  const now = Date.now();
  const start = new Date(now - TIME_WINDOW);
  const end = new Date(now);
  
  return {
    start: `"${start.toISOString()}"`,
    end: `"${end.toISOString()}"`
  };
}
```